src/irmin-pack/unix/ext.ml

```diff
@@ -22,6 +22,16 @@ module Maker (Config : Conf.S) = struct
   include Pack_key.Store_spec
 
   module Make (Schema : Irmin.Schema.Extended) = struct
+    (* Register functions to be called when process terminates. If there is a gc
+       process running, make sure to terminate it. *)
+    let at_exit_kill_subprocess = ref None
+
+    let () =
+      at_exit (fun () ->
+          match !at_exit_kill_subprocess with
+          | None -> ()
+          | Some pid -> Unix.kill pid Sys.sigkill)
```

Comment:
This looks good. I would however:
1. move it to the top of the file so that we don't call `at_exit` a thousand time if a user creates a thousand pack stores;
2. turn it to a list so that we support the very unlikely event of multiple concurrent GC from different stores;
3. protect the list with an `Stdlib.Mutex.t`.

Reply:
I tried to address your comments in separate commits, let me know if this looks ok to you.